app/validators/not_forbidden_name_validator.rb

```diff
@@ -0,0 +1,8 @@
+class NotForbiddenNameValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.nil?
+    return unless ArchiveConfig.FORBIDDEN_USERNAMES.include?(value.downcase)
+
+    record.errors.add(attribute, options[:message] || I18n.t("validators.forbidden_name", value: value))
```

Comment:
Unlike the cases above, it's okay to use `t` instead of adding a symbolic error here, but I do think I'd prefer a symbolic error — it makes it easier to customize the message for different classes. (I used the `BlacklistedEmailValidator` as an example to follow not because I like the way it's written, but because it was the validator with the closest functionality.)

Still, this change is more aesthetic than anything, since currently we don't need to customize the error messages on a per-class basis. So we can leave the code as is if you like it this way.

Reply:
I also like the symbol way better, and I probably should've done a more thorough survey of those validator classes first. Updated!